subsys/bluetooth/host/long_work.h

```diff
@@ -0,0 +1,20 @@
+/* long_work.h - Workqueue intended for long-running operations. */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __LONG_WORK_H_
+#define __LONG_WORK_H_
+
+#if defined(CONFIG_BT_LONG_WQ)
+struct k_work_q * bt_long_wq_get(void);
+#else
+#include <zephyr/kernel.h>
+
+inline struct k_work_q * bt_long_wq_get(void)
+{
+	return &k_sys_work_q;
+}
```

Comment:
Hmm, not sure how I feel about this, as `bt_long_wq_get` might suggest that it is a specific WQ. 

Could there not be a case where you'd want to either use the long_wq (if enabled), or a workqueue different from the system workqueue in case that the long_wq isn't available?

Reply:
what parameter type would you propose ? I thought of `prio` but am not sure.

Reply:
I am not sure `bt_long_wq_get` should return `&k_sys_work_q` at all. IMO, since we added a Kconfig for it, `bt_long_wq_get` should be callable if, and only if, the long workqueue Kconfig is enabled.